server/internal/adapter/gql/gqlmodel/convert.go

```diff
@@ -45,3 +51,22 @@ func ToPagination(pagination *Pagination) *usecasex.Pagination {
 		Last:   pagination.Last,
 	}
 }
+
+func FromFile(f *graphql.Upload) *file.File {
+	if f == nil {
+		return nil
+	}
+	return &file.File{
+		Content:     io.NopCloser(f.File),
+		Path:        ToGCNaming(f.Filename),
+		Size:        f.Size,
+		ContentType: f.ContentType,
+	}
+}
+
+func ToGCNaming(s string) string {
+	// TODO: we need a better way to implement google storage naming
+	re := regexp.MustCompile(`[^a-zA-Z0-9-.]`)
+	ss := re.ReplaceAllString(s, "-")
+	return strings.ToLower(ss)
+}
```

Comment:
```suggestion
```

This function should not be located here as it relates to the internals of GCS. There seems to be an error in the implementation policy when knowledge of GCS appears in this package.

Reply:
Where should this function be located? can you explain more please?